docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+
+* **All-in-one** pods which can perform the ping check for all connectors. These would have access to all connectors supported and expose a service which has the connector type as parameter as well as the `ConnectorConfiguration` for calling out to the connector. A single K8s service is in front of these pods which is called by the `ipaas-rest`. This will work only if all connectors are known upfront or connectors could be loaded dynamically. 
+* **Dedicated** pods which holds only one connector each. Each of this pod has a different service as entry point (whose name is associated with the connector name). So there is one pod, one service and `ipaas-rest` will discover the proper service (by naming convention to the connector name) and call it with the `ConnectorConfiguration`. The advantage is, that if no such service could be found, the `ipaas-rest` could install the service on the fly. This works also for connectors which are not known upfront but might be defined later (e.g by expert developers).
+* **Mixed** pods which hold a set of connectors but also know how to dynamically load new connectors. For each connector it add a label or annotation to this pod. This set of labels (where the connector type is part of the key name) can be queried from the outside. Services can be defined to select on these labels, so that there could be also one service per connector which has a single label as selector matching all pods which have this connector available. Again, if there is no matching service or the service does not match any pod, then a new pod can be fired up dynamically, containing the requested connector. The main point here is, that the connector types are attached as labels, e.g. `connector.twitter.query=1, connector.salesforce.create-contact=1`, ... This is probably the most flexible one as it can be also scaled independently. The challenge is how to load connectors dynamically from a central (Nexus) repository, how to update the Pods meta-data and how to dynamically create services.
+
+On a first glance, the fist **all-in-one** option is the easiest one, as it only needs a single service, where the pods behind will either already have all connectors preloaded or can load dynamically.
+
+## Open Questions
+
+* Is the set of connectors static or can these be defined by _expert developers_, too ?
+* If connectors can be defined dynamically, how ...
+    * ... are these build ?
+    * ... are they distributed ? (via Maven repo ?)
+    * ... how can they be loaded by the tester pods ?
+* How should I18N works for the error messages returned ? Is each connector responsible for I18N (and the locale should be part of the API) ? Or is I18N out of scope ?
+* Can connectors be tested with out a Camel route ? (would be the easiest)
+
+## API suggestion
+
+```java
+interface Pingable {
```

Comment:
Yes, makes absolutely sense to have it on the `Component` interface, probably differentiating between producer and consumer (since you might only have read-only access to an endpoint).